fhir-persistence-schema/docs/DropTenant.md

```diff
@@ -0,0 +1,129 @@
+# Removing a tenant from the database
+
+Use this guide when removing a tenant from the Db2 multi-tenant database. The tenant's data will be removed from the database along with any associated storage.
+
+To obtain a list of the tenants currently managed in a given database, run the following:
+
+```sh
+    java -jar schema/fhir-persistence-schema-*-cli.jar \
+      --prop-file db2.properties --list-tenants
+```
+
+```
+ TenantId     Status       TenantName Schema
+        1  ALLOCATED            test1 FHIRDATA
+        2  ALLOCATED            test2 FHIRDATA2
+        3    DROPPED            test3 <not-known>
+```
+
+Note that after the tenant is dropped, it is no longer associated with a schema.
+
+## Background
+
+Tenant data is stored in a Db2 table partition attached to each of the IBM FHIR Server data tables. Removal occurs in two phases:
+
+- **Phase 1**. The tenant's partitions are detached from each data table and converted into a stand-alone table.
+- **Phase 2**. The stand-alone tables are dropped. Once all the tables are dropped, the tablespace will also be dropped.
```

Comment:
just curious, could a tenant be partially dropped? e.g, what if the tool fails to finish all steps for whatever reason.

Reply:
Yes, that's why I suppress the exceptions, to allow it to be run multiple times. It will only succeed once you can drop the table space (and that exception isn't suppressed). So keep running phase 2 until you don't get an error. Phase 1 includes phase 2, so you could just keep running phase 1...but not recommended, because of all the FK disable/enable stuff that also goes on there. It would extend any maintenance window.